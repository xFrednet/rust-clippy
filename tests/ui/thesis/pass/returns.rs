//@rustc-env: CLIPPY_PETS_PRINT=1
//@rustc-env: CLIPPY_PRINT_MIR=1

#![warn(clippy::borrow_pats)]

fn main() {}

fn simple_const() -> u32 {
    15
}

fn fn_call() -> u32 {
    simple_const()
}

fn simple_cond() -> u32 {
    if true { 1 } else { 2 }
}

fn static_slice() -> &'static [u32] {
    &[]
}

fn static_string() -> &'static str {
    "Ducks are cool"
}

fn arg_or_default(arg: &String) -> &str {
    if arg.is_empty() { "Default" } else { arg }
}
