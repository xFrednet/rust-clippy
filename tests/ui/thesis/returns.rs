fn simple_const() -> u32 {
    15
}

#[forbid(clippy::borrow_pats)]
fn fn_call() -> u32 {
    simple_const()
}

fn simple_cond() -> u32 {
    if true { 1 } else { 2 }
}

fn static_slice() -> &'static [u32] {
    &[]
}

fn main() {}
