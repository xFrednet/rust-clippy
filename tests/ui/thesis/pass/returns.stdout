# "main"
- Body: Output(Unit)     , NotConst , Safe  , Sync , Free       {NoArguments}

# "simple_const"
- Body: Output(Primitive), NotConst , Safe  , Sync , Free       {NoArguments}

# "fn_call"
- Body: Output(Primitive), NotConst , Safe  , Sync , Free       {NoArguments}

# "simple_cond"
- Body: Output(Primitive), NotConst , Safe  , Sync , Free       {NoArguments}

# "static_slice"
- Body: Output(Reference), NotConst , Safe  , Sync , Free       {NoArguments}

# "static_string"
- Body: Output(Reference), NotConst , Safe  , Sync , Free       {NoArguments}

# "arg_or_default"
- Body: Output(Reference), NotConst , Safe  , Sync , Free       {ReturnedStaticLoanForNonStatic}

